changelogs/unreleased/gh-3594-fail-safely-if-tuple-limit-is-reached.md

```diff
@@ -0,0 +1,4 @@
+## bugfix/core
+
+* Fixed a potential data corruption when the number of tuples in a hash index
```

Comment:
Don't we have the same problem with the tree index?

Reply:
Yes, but the limit for the tree index is way bigger:
- https://github.com/tarantool/tarantool/issues/3594#issuecomment-576360867